coderd/workspaces.go

```diff
@@ -165,10 +165,37 @@ func (api *API) workspaceByOwnerAndName(rw http.ResponseWriter, r *http.Request)
 	owner := httpmw.UserParam(r)
 	workspaceName := chi.URLParam(r, "workspacename")
 
+	var (
+		includeDeletedStr = r.URL.Query().Get("include_deleted")
+		includeDeleted    = false
+	)
+	if includeDeletedStr != "" {
+		var err error
+		includeDeleted, err = strconv.ParseBool(includeDeletedStr)
+		if err != nil {
+			httpapi.Write(rw, http.StatusBadRequest, httpapi.Response{
+				Message: fmt.Sprintf("Invalid boolean value %q for \"include_deleted\" query param.", includeDeletedStr),
+				Validations: []httpapi.Error{
+					{Field: "include_deleted", Detail: "Must be a valid boolean"},
+				},
+			})
+			return
+		}
+	}
+
 	workspace, err := api.Database.GetWorkspaceByOwnerIDAndName(r.Context(), database.GetWorkspaceByOwnerIDAndNameParams{
 		OwnerID: owner.ID,
 		Name:    workspaceName,
 	})
+
+	if includeDeleted && errors.Is(err, sql.ErrNoRows) {
+		workspace, err = api.Database.GetWorkspaceByOwnerIDAndName(r.Context(), database.GetWorkspaceByOwnerIDAndNameParams{
+			OwnerID: owner.ID,
+			Name:    workspaceName,
+			Deleted: includeDeleted,
```

Comment:
I don't think this works as intended. The param here in the SQL matches for `deleted = $includeDeleted`.

So if you set this to true, the workspace will not show if the workspace is deleted. We will need to update the SQL too, or just always return deleted workspaces by default.

Reply:
See how it's handled here: https://github.com/coder/coder/blob/454aea7f85dcefb736789cb7e47872e62a19401c/coderd/workspaces.go#L57-L62

I think the `deleted` query param is kinda annoying as you have to know the state of the workspace to query it.

Reply:
@Emyrk Hmm, testing it in the UI seems to work.

1. if `include_deleted` is false, then we never enter this code block (because `errors.Is(err, sql.ErrNoRows)` is false
2. otherwise we enter this code block and try to query for a deleted workspace

In either case, if we get nothing back, we then go into the `httpapi.Forbidden(rw)` block below. I agree editing the SQL would also work. I just didn't know what other ramifications that would have.  

Let me know if I'm misunderstanding your comment!